File: Services/QuestionService.cs
using Dapper;
using Microsoft.Extensions.Options;
using System;
using System.Data;
using System.Linq;
using Microsoft.Data.SqlClient;
using WebApi.Helpers;
using WebApi.Models;
using WebApi.Services.Interfaces;

namespace WebApi.Services
{
    public class QuestionService : IQuestionService
    {
        private readonly ConnectionStrings _connectionStrings;

        public QuestionService(IOptions<ConnectionStrings> connectionStrings)
        {
            _connectionStrings = connectionStrings.Value;
        }

        public dynamic Select(QuestionFilterModel filtro)
        {
            using (var con = new SqlConnection(_connectionStrings.Default))
            {
                try
                {
                    con.Open();
                    var ret = con.Query("SP_sl_Question", new
                    {
                        filtro.QuestionId,
                        filtro.IsComplement,
                        filtro.Rank,
                        filtro.Question,
                        filtro.AnswerTypeCd,
                        filtro.Group,
                        filtro.Metric,
                        filtro.IsFirstSurvey,
                        filtro.IsFinalSurvey,
                        filtro.IsCompetenceLevel,
                        filtro.IsFinishEarly,
                        filtro.IsStandardMetric,
                        filtro.IsSglYesNoType,
                        filtro.IsFeedback
                    }, commandType: CommandType.StoredProcedure).ToList();
                    return ret;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                finally
                {
                    con.Close();
                }
            }
        }

        public dynamic Create(QuestionCreateModel model)
        {
            using (var con = new SqlConnection(_connectionStrings.Default))
            {
                try
                {
                    con.Open();
                    var ret = con.Query("SP_cr_Question", new
                    {
                        model.IsComplement,
                        model.Rank,
                        model.Question,
                        model.Description,
                        model.AnswerTypeCd,
                        model.Group,
                        model.Metric,
                        model.IconMetric,
                        model.IsFirstSurvey,
                        model.IsFinalSurvey,
                        model.IsCompetenceLevel,
                        model.IsFinishEarly,
                        model.IsStandardMetric,
                        model.IsSglYesNoType,
                        model.IsFeedback
                    }, commandType: CommandType.StoredProcedure).ToList();
                    return ret;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                finally
                {
                    con.Close();
                }
            }
        }

        public dynamic Update(QuestionUpdateModel model)
        {
            using (var con = new SqlConnection(_connectionStrings.Default))
            {
                try
                {
                    con.Open();
                    var ret = con.Query("SP_up_Question", new
                    {
                        model.QuestionId,
                        model.IsComplement,
                        model.Rank,
                        model.Question,
                        model.Description,
                        model.AnswerTypeCd,
                        model.Group,
                        model.Metric,
                        model.IconMetric,
                        model.IsFirstSurvey,
                        model.IsFinalSurvey,
                        model.IsCompetenceLevel,
                        model.IsFinishEarly,
                        model.IsStandardMetric,
                        model.IsSglYesNoType,
                        model.IsFeedback
                    }, commandType: CommandType.StoredProcedure).ToList();
                    return ret;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                finally
                {
                    con.Close();
                }
            }
        }

        public dynamic Delete(int questionId)
        {
            using (var con = new SqlConnection(_connectionStrings.Default))
            {
                try
                {
                    con.Open();
                    var ret = con.Query("SP_dl_Question", new
                    {
                        QuestionId = questionId
                    }, commandType: CommandType.StoredProcedure).ToList();
                    return ret;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                finally
                {
                    con.Close();
                }
            }
        }
    }
}

File: portal-supervisao-360/src/app/core/services/question.service.ts
import { Injectable } from '@angular/core';
import { HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { QuestionModel, QuestionCreateModel, QuestionUpdateModel } from '../../shared/models/survey-admin.models';

export interface QuestionFilter {
  questionId?: number;
  isComplement?: boolean;
  rank?: number;
  question?: string;
  answerTypeCd?: string;
  group?: string;
  metric?: string;
  isFirstSurvey?: boolean;
  isFinalSurvey?: boolean;
  isCompetenceLevel?: boolean;
  isFinishEarly?: boolean;
  isStandardMetric?: boolean;
  isSglYesNoType?: boolean;
  isFeedback?: boolean;
  surveyTypeId?: number;
}

@Injectable({ providedIn: 'root' })
export class QuestionService {
  constructor(private api: ApiService) {}

  select(filtro: QuestionFilter = {}): Observable<QuestionModel[]> {
    let params = new HttpParams();
    Object.entries(filtro).forEach(([k, v]) => {
      if (v != null) {
        params = params.set(k, String(v));
      }
    });
    return this.api.getWithParams<QuestionModel[]>('/Question/Select', params);
  }

  create(model: QuestionCreateModel): Observable<unknown> {
    return this.api.post('/Question/Create', model);
  }

  update(model: QuestionUpdateModel): Observable<unknown> {
    return this.api.put('/Question/Update', model);
  }

  delete(id: number): Observable<unknown> {
    return this.api.delete(`/Question/Delete/${id}`);
  }
}
